File: okj/rendszerüzemeltetö/dharomszog/DHaromszog.java
public class DHaromszog{
    private double aOldal;
    private double bOldal;
    private double cOldal;
    private int sorszam;
    
    public DHaromszog(String sor, int sorszam) throws OldalNullaException, AdatNemJoSorrendException, NemSzerkeszthetoException, NemDerekszoguException{
        var oldalak = sor.replace(',', '.').split(" ");
        
        var beAOldal = Double.parseDouble(oldalak[0]);
        var beBOldal = Double.parseDouble(oldalak[1]);
        var beCOldal = Double.parseDouble(oldalak[2]);
        
        setaOldal(beAOldal);
        setbOldal(beBOldal);
        setcOldal(beCOldal);
        
        if(!isNovekvoSorrend()) {
            throw new AdatNemJoSorrendException();
        }
        
        if(!isSzerkesztheto()) {
            throw new NemSzerkeszthetoException();
        }
        
        if(!isDerekszogu()) {
            throw new NemDerekszoguException();
        }
        
        this.sorszam = sorszam;
    }
    
    private boolean isNovekvoSorrend() {
        return aOldal <= bOldal && bOldal <= cOldal;
    }
    
    private boolean isSzerkesztheto() {
        return (aOldal + bOldal) > cOldal;
    }
    
    private boolean isDerekszogu() {
        return (cOldal * cOldal) == (aOldal * aOldal + bOldal * bOldal);
    }
    
    public double kerulet() {
        return aOldal + bOldal + cOldal;
    }
    
    public double terulet() {
        return aOldal * bOldal / cOldal;
    }

    public double getaOldal() {
        return aOldal;
    }

    public double getbOldal() {
        return bOldal;
    }

    public double getcOldal() {
        return cOldal;
    }
    
    public int getSorszam() {
        return sorszam;
    }

    public void setSorszam(int sorszam) {
        this.sorszam = sorszam;
    }
    
    public void setaOldal(double aOldal) throws OldalNullaException{
        if(aOldal > 0D) {
            this.aOldal = aOldal;
        }else{
            throw new OldalNullaException("Az \"a\" oldal nem lehet nulla vagy negatív!");
        }
    }
    
    public void setbOldal(double bOldal) throws OldalNullaException{
        if(aOldal > 0D) {
            this.bOldal = bOldal;
        }else {
            throw new OldalNullaException("A \"b\" oldal nem lehet nulla vagy negatív!");
        }
    }
    
    public void setcOldal(double cOldal) throws OldalNullaException{
        if(aOldal > 0D) {
            this.cOldal = cOldal;
        }else {
            throw new OldalNullaException("A \"c\" oldal nem lehet nulla vagy negatív!");
        }
    }
}

File: okj/rendszerüzemeltetö/dharomszog/c#/DHaromszog.cs
using System;

public class DHaromszog {
    private double _aOldal;
    private double _bOldal;
    private double _cOldal;

    public DHaromszog(string sor, int sorszam) {
        var oldalak = sor.Split(' ');

        Console.WriteLine(String.Join('\n', oldalak));

        aOldal = double.Parse(oldalak[0]);
        bOldal = double.Parse(oldalak[1]);
        cOldal = double.Parse(oldalak[2]);

        if(!NovekvoSorrendE()) {
            throw new AdatNemJoSorrendException();
        }

        if(!SzerkeszthetoE()) {
            throw new NemSzerkeszthetoException();
        }

        if(!DerekszoguE()) {
            throw new NemDerekszoguException();
        }

        this.Sorszam = sorszam;
    }

    public int Sorszam { get; set; }
    public double aOldal { get => this._aOldal;
                           set {
                               if(value > 0D) {
                                    this._aOldal = value;
                                }else{
                                    throw new OldalNullaException("Az \"a\" oldal nem lehet nulla vagy negatív!");
                                }
                           }}

    public double bOldal { get => this._bOldal;
                           set {
                               if(value > 0D) {
                                    this._bOldal = value;
                                }else {
                                    throw new OldalNullaException("A \"b\" oldal nem lehet nulla vagy negatív!");
                                }
                           }}

    public double cOldal { get => this._cOldal;
                           set {
                                if(value > 0D) {
                                    this._cOldal = value;
                                }else {
                                    throw new OldalNullaException("A \"c\" oldal nem lehet nulla vagy negatív!");
                                }
                           }}

    private bool NovekvoSorrendE() => aOldal <= bOldal && bOldal <= cOldal;
    private bool SzerkeszthetoE() => (aOldal + bOldal) > cOldal;
    private bool DerekszoguE() => (cOldal * cOldal) == (aOldal * aOldal + bOldal * bOldal);
    public double Kerulet() => aOldal + bOldal + cOldal;
    public double Terulet() => aOldal * bOldal / cOldal;
}